app/waf/.vscode/launch.json

```diff
@@ -1,14 +1,46 @@
 {
   "version": "0.2.0",
   "configurations": [
+    {
+      "type": "node",
+      "request": "launch",
+      "name": "Mocha Tests",
+      "cwd": "${workspaceFolder}",
+      "program": "${workspaceFolder}/node_modules/mocha/bin/_mocha",
+      "args": [
+        "--colors",
+        "--no-timeouts",
+        "--debug-brk",
+        //"${workspaceFolder}/dist/test/<your specific file>.js",
+        //"${workspaceFolder}/dist/test/**/*.js",
+        "${workspaceFolder}/dist/test/acceptance/openstack.controller.acceptance.js"
+      ],
+      "runtimeExecutable": null,
+      "internalConsoleOptions": "openOnSessionStart",
+      "env": {
+        "NODE_ENV": "production"
+      },
+      "stopOnEntry": false,
+      "preLaunchTask": "Single Build"
+    },
     {
       "name": "Launch",
       "type": "node",
       "request": "launch",
       "program": "${workspaceFolder}/index.js",
       "args": [],
       "cwd": "${workspaceFolder}",
-      "protocol": "inspector"
+      "protocol": "inspector",
+      "env": {
+        "TEST_OPENSTACK_INTEGRATION": "1"
+        // "PRODUCT_RELEASE": "1",
```

Comment:
Now I prefer to remove .vscode from repo, after looking this... since it benefit very little to other contributors. We should encourage the diversity among contributors. Everyone can manage their ow editor and configuration file by themself, instead of mange them in a shared repo.

Reply:
This may draw another topic: I recommend that we use the very same developing environment. VS/VSCode is the recommended IDE from MS for typescript.
However, not forced. It's OK to remove it from the code repo, personally, I'll keep it myself. 


It's OK to remove it from the code repo, personally, I'll keep it myself.